06week/higherOrder.js

```diff
@@ -2,17 +2,52 @@
 
 const assert = require('assert');
 
-function forEach(arr, callback) {
-  // Your code here
+// Create a forEach() function that takes an array of items and 
+// a function that runs the function arr.length number of times.
+
+const arr = ['arya','jon','danaerys','brianne'];
+
+const forEach = (arr, callback)=>{
+  for(let i = 0 ; i< array1.length ; i++ ){
+    callback(arr[i])
+  }
 }
 
-function map(arr, callback) {
-  // Your code here
+// Create a map() function that takes an array of items and a function 
+// that returns an array with each item manipulated by that function.
+
+const ar = [1, 2, 3, 4, 5];
+const empty = [];
```

Comment:
Your array empty should be inside your map function. Although, this does work because it is a global variable.